web/packages/teleport/src/Discover/Server/CreateEc2Ice/CreateEc2Ice.tsx

```diff
@@ -0,0 +1,192 @@
+  Header,
+  SecurityGroupPicker,
+} from 'teleport/Discover/Shared';
+
+import { CreateEc2IceDialog } from './CreateEc2IceDialog';
+
+type TableData = {
+  items: SecurityGroup[];
+  nextToken?: string;
+  fetchStatus: FetchStatus;
+};
+
+export function CreateEc2Ice() {
+  const [showCreatingDialog, setShowCreatingDialog] = useState(false);
+  const [selectedSecurityGroups, setSelectedSecurityGroups] = useState<
+    string[]
+  >([]);
+  const [tableData, setTableData] = useState<TableData>({
+    items: [],
+    nextToken: '',
+    fetchStatus: 'disabled',
+  });
+
+  function onSelectSecurityGroup(
+    sg: SecurityGroup,
+    e: React.ChangeEvent<HTMLInputElement>
+  ) {
+    if (e.target.checked) {
+      return setSelectedSecurityGroups([...selectedSecurityGroups, sg.id]);
+    } else {
+      setSelectedSecurityGroups(
+        selectedSecurityGroups.filter(id => id !== sg.id)
+      );
+    }
+  }
+
+  useEffect(() => {
+    fetchSecurityGroups();
+  }, []);
+
+  const {
+    attempt: fetchSecurityGroupsAttempt,
+    setAttempt: setFetchSecurityGroupsAttempt,
+  } = useAttempt('');
+
+  const { attempt: deployEc2IceAttempt, setAttempt: setDeployEc2IceAttempt } =
+    useAttempt('');
+
+  const { emitErrorEvent, agentMeta, prevStep, nextStep } = useDiscover();
+
+  async function fetchSecurityGroups() {
+    const integration = (agentMeta as NodeMeta).integration;
+
+    setFetchSecurityGroupsAttempt({ status: 'processing' });
+    try {
+      const { securityGroups, nextToken } =
+        await integrationService.fetchSecurityGroups(integration.name, {
+          vpcId: (agentMeta as NodeMeta).node.awsMetadata.vpcId,
+          region: (agentMeta as NodeMeta).node.awsMetadata.region,
+          ...(tableData.nextToken && { nextToken: tableData.nextToken }),
+        });
+
+      setFetchSecurityGroupsAttempt({ status: 'success' });
+      setTableData({
+        nextToken: nextToken,
+        fetchStatus: nextToken ? '' : 'disabled',
+        items: [...tableData.items, ...securityGroups],
+      });
+    } catch (err) {
+      const errMsg = getErrMessage(err);
+      setFetchSecurityGroupsAttempt({ status: 'failed', statusText: errMsg });
+      emitErrorEvent(`fetch security groups error: ${errMsg}`);
+    }
+  }
+
+  async function deployEc2InstanceConnectEndpoint() {
+    const integration = (agentMeta as NodeMeta).integration;
+
+    setDeployEc2IceAttempt({ status: 'processing' });
+    setShowCreatingDialog(true);
+    try {
+      await integrationService.deployAwsEc2InstanceConnectEndpoint(
+        integration.name,
+        {
+          region: (agentMeta as NodeMeta).node.awsMetadata.region,
+          subnetId: (agentMeta as NodeMeta).node.awsMetadata.subnetId,
+          ...(selectedSecurityGroups.length && {
+            securityGroupIds: selectedSecurityGroups,
+          }),
+        }
+      );
+      // Capture event for deploying EICE.
+      // emitEvent(null); TODO rudream (ADD EVENTS FOR EICE FLOW)
```

Comment:
just fyi, some of these are done under the hood, unless you specify `manuallyEmitSuccessEvent` in the `Discover/Server/index.ts` config file

Reply:
We want both, we want to emit the event for deploying EICE manually, and then also for the createNode event to be emitted automatically under the hood when they go to the next step, I made these changes in this PR: https://github.com/gravitational/teleport/pull/32617